subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AnnotationProcessorDeclaration.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile.processing;
+
+/**
+ * Information about an annotation processor, based on its static metadata
+ * in META-INF/services/javax.annotation.processing.Processor.
+ */
+public class AnnotationProcessorDeclaration {
```

Comment:
I guess you have this class because you anticipate more things to be stored in there, like the fact it's incremental or not. I'd suggest to add it to the docs.

Reply:
Yes, this is going to be added in the next batch of changes.